src/vipersci/nss.py

```diff
@@ -119,6 +110,72 @@ def __call__(
         return d1, d2
 
 
+class NssModeler:
```

Comment:
So this would be called as `nss.NssModeler`.  I'd prefer that this have a parallel naming to the DataSimulator class, and that this would be DataModeler (and thus `nss.DataModeler`)